File: Data Structures & Algorithms/climbing-stairs/submission-2.cpp
class Solution {
public:
int func(int n,vector<int> &dp){
    if(n<=1) return 1;
    if(dp[n]!=-1) return dp[n];
    return dp[n]=func(n-2,dp)+func(n-1,dp);
}
    int climbStairs(int n) {
        vector<int> dp(n+1,-1);
        int ans=func(n,dp);
        return ans;
        
    }
};

File: Data Structures & Algorithms/partition-labels/submission-2.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        last_occurence={}
        for i,e in enumerate(s) :
            last_occurence[e]=i
        ans=[]
        farthest=0
        start=0
        for i,e in enumerate(s):
            farthest=max(farthest,last_occurence[e])
            if i==farthest:
                ans.append(farthest-start+1)
                start=i+1
        return ans
        
        